src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets

```diff
@@ -272,50 +243,30 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
-                                        _FilterOutFXFromResolvedCandidates
-    ============================================================
-    -->
-  
-  <Target Name="_FilterOutFXFromResolvedCandidates"
-          Condition="$(SkipRemovingSystemFiles) != 'true' ">
-
-    <ItemGroup>
-      <_ManagedResolvedFilesToOptimize Include="@(_ManagedResolvedFileToPublishCandidates)" Condition="!Exists('$(_NetCoreRefDir)$(DirectorySeparatorChar)%(FileName)%(Extension)')"/>
-    </ItemGroup>
-  </Target>
-
-  <!--
-    ============================================================
-                                        _PopulateUnFilteredResolvedCandidates
-    ============================================================
-    -->
-  <Target Name="_PopulateUnFilteredResolvedCandidates"
-          Condition="$(SkipRemovingSystemFiles) == 'true' ">
-
-   <ItemGroup>
-      <_ManagedResolvedFilesToOptimize Include="@(_ManagedResolvedFileToPublishCandidates)" />
-    </ItemGroup>
-  </Target>
-
-  <!--
-    ============================================================
                                         _RestoreCrossgen
 Restores netcoreapp and publishes it to a temp directory
     ============================================================
     -->
   
   <Target Name="_RestoreCrossgen"
-           Condition="$(SkipOptimization) != 'true' ">
-    <MSBuild Projects="$(_CrossProjFile)"
+          DependsOnTargets="PrepforRestoreForComposeStore;
```

Comment:
I think dependencies like this make it hard to reason about the flow.  ComposeStore depends on CrossGen, but here CrossGen is depending back on ComposeStore targets.  It would be nice if there was a clear graph of dependencies between these two sets of targets.